src/DataTypes/transformTypesRecursively.cpp

```diff
@@ -47,6 +47,7 @@ void transformTypesRecursively(DataTypes & types, std::function<void(DataTypes &
         bool have_tuple = false;
         bool all_tuples = true;
         size_t tuple_size = 0;
+        bool sizes_are_equal = true;
```

Comment:
I know it's not connected to this change, but can we get stack overflow with this function if someone for example generates a JSON with a lot of nesting?

Reply:
Yes, it's possible. Good catch. I will add a setting to limit the depth.